app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision_workflow.rb

```diff
@@ -22,8 +22,53 @@ def volume_dialog_keys
     %i[name size shareable]
   end
 
+  def vm_image
+    template_id = values&.dig(:src_vm_id, 0)
+    ar_ems.miq_templates.select { |t| t.id == template_id }[0]
```

Comment:
Might want to memoize this as `@vm_image ||= ` since it looks like you reference `sap_image?` and `sap_flavor` a few times.

Also `ems.miq_templates.select { |t| t.id == template_id }[0]` looks like just `ems.miq_templates.find_by(:id => template_id)`